ssl/statem/statem_dtls.c

```diff
@@ -49,76 +49,57 @@ static int dtls_get_reassembled_message(SSL_CONNECTION *s, int *errtype,
 
 static dtls_sent_msg *dtls1_sent_msg_new(size_t msg_len)
 {
-    dtls_sent_msg *msg = NULL;
-    unsigned char *msg_buf = NULL;
+    dtls_sent_msg *msg = OPENSSL_malloc(sizeof(*msg) + msg_len);
 
-    if ((msg = OPENSSL_zalloc(sizeof(*msg))) == NULL)
+    if (msg == NULL)
         return NULL;
 
-    if (msg_len) {
-        if ((msg_buf = OPENSSL_malloc(msg_len)) == NULL) {
-            OPENSSL_free(msg);
-            return NULL;
-        }
-    }
+    memset(msg, 0, sizeof(*msg));
 
     /* zero length msg gets msg->msg_buf == NULL */
-    msg->msg_buf = msg_buf;
+    if (msg_len > 0)
+        msg->msg_buf = (unsigned char *)(msg + 1);
 
     return msg;
 }
 
 void dtls1_sent_msg_free(dtls_sent_msg *msg)
 {
-    if (msg == NULL)
-        return;
-
-    OPENSSL_free(msg->msg_buf);
-    OPENSSL_free(msg);
+    if (msg != NULL)
+        OPENSSL_free(msg);
```

Comment:
OPENSSL_free already does the null check, could save the conditional here I think